nemo/deploy/query.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from .utils import str_list2numpy
+
+use_pytriton = True
+try:
+    from pytriton.client import DecoupledModelClient, ModelClient
+except:
+    use_pytriton = False
+
+
+class NemoQueryBase(ABC):
```

Comment:
Please do not use vague names like NeMoBaseQuery and also specialize on query_llm as methods. This is at the level of the NeMo core - that is it cannot specialize onto any domain. You can have this class act as a base class and have subclasses that specialize on LLM vs whatever else with appropriate name such as NeMoLLMBaseQuery.

Also the name `NeMoBaseQuery` is not indicative of what it is doing. Please come up with a better name, something that denotes it is a wrapper for a DeployBase with the Pytriton backend (and can potentially have other backends).

Reply:
I agree and changed the class names to NemoQueryLLMBase and NemoQueryLLM, and file name to query_llm.

NemoBaseQuery is actually not a wrapper for pytriton backend. It's just a class to send LLM specific query to a server like Triton. It's using pytriton classes to send the query but it really doesn't have to. Just used it for convenience.